src/mono/mono/eglib/glib.h

```diff
@@ -1316,4 +1310,424 @@ gint
 g_clock_nanosleep (clockid_t clockid, gint flags, const struct timespec *request, struct timespec *remain);
 #endif
 
+//#define ENABLE_CHECKED_CASTS
+#ifdef ENABLE_CHECKED_CASTS
```

Comment:
@steveisok should we have some checked cast CI runs?

Reply:
I think we probably need to work that through locally first, maybe after we are done with the warnings? When enabled (and I implemented asserts in the inline functions to validate underflow/overflow), we hit underflow/overflows that appears to be intentional in some cases, so if we would like to assert, we would probably need to do local test run and replace use of macro with direct cast where the intention is to underflow/overflow (and maybe do explicit comment that it's the case) or investigate if these are actual real bugs that needs to be addressed.